src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -601,13 +611,13 @@ private static bool StringRequiresHtmlDecoding(string s)
 
         #endregion
 
-        #region UrlDecoder nested class
+        #region UrlDecoder nested struct
 
         // *** Source: alm/tfs_core/Framework/Common/UriUtility/HttpUtility.cs
         // This specific code was copied from above ASP.NET codebase.
 
-        // Internal class to facilitate URL decoding -- keeps char buffer and byte buffer, allows appending of either chars or bytes
-        private class UrlDecoder
+        // Internal struct to facilitate URL decoding -- keeps char buffer and byte buffer, allows appending of either chars or bytes
```

Comment:
While you're here, can you delete the above comment about the original source for this?